Automattic/jetpack
# Admin menu: Change order of Jetpack sub-menu items

Fixes https://github.com/Automattic/jetpack-roadmap/issues/856

Continuation of #38768

## Proposed changes:

* This PR changes the order of the Jetpack sub-menu items, based on this issue comment: https://github.com/Automattic/jetpack-roadmap/issues/856#issuecomment-2308599496.

<details>
<summary>View the order of items here</summary>
My Jetpack<br/>
Stats<br/>
Boost<br/>
VideoPress<br/>
Social<br/>
Protect<br/>
Akismet Anti-spam<br/>
VaultPress Backup<br/>
Activity Log (External)<br/>
Scan<br/>
Search<br/>
Subscribers (External)<br/>
WordAds<br/>
Settings<br/>
Dashboard<br/>
Jetpack Manage (External)<br/>
Beta Tester<br/>
</details>

* This PR also updates the logic to ensure that Dashboard, Settings, and Activity Log items are shown as selected, when the relevant dashboard pages are open.
* Other Jetpack menu items apart from those mentioned above haven't been considered here, for example Monetize on WoA sites.
* Also, ideally we'd add tests here, particularly for `jetpackPageOrder` (ensuring the correct menu item is showing as selected). Since there is a level of urgency in getting this change out I'll follow up separately: https://github.com/Automattic/vulcan/issues/482
* Simple sites have their own order defined, see also `add_jetpack_submenu` in `mu-plugins/jetpack-newsletter.php`

### Other information:


## Jetpack product discussion

See https://github.com/Automattic/jetpack-roadmap/issues/856

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

This is relevant for self-hosted and WoA sites - Simple sites override menu locations.

* With this PR applied, you should be able to see Jetpack menu items in the same order as explained under Proposed Changes above. 
* The Dashboard and Settings menu items should show as selected when clicked, regardless of the number of menu items before or after them.
* To test with all menu items, note that Scan can be added by adding a Scan plan (and ensure Scan is working on the site) and deactivating Protect. WordAds can be added by including `define( 'ENABLE_WORDADS_SHARED_UI', true );` in your `wp-config.php` file. Subscribers will be added if you enable the Newsletter feature at `/wp-admin/admin.php?page=jetpack#/newsletter`. I found the best way to add the other items if not testing locally is using the Jetpack Beta tester plugin, then activating all relevant plugins. However as Akismet is not included with the Beta tester plugin, you'd have to apply the branch for a plugin that uses the admin-ui package, which is most (eg. Jetpack, Beta plugin, Boost..).
* You'll also want to test without Jetpack installed as well, and just some standalones instead. If testing on Jurassic Ninja or WoA, using the Beta tester plugin is the best way to do this as well, as you'll need to ensure you're applying this branch to the relevant standalone to make sure the menu priority is up to date - including Jetpack Beta itself.
